sdk/keyvault/azure-keyvault-administration/tests/test_access_control.py

```diff
@@ -37,27 +49,60 @@ def get_service_principal_id(self):
-            assert definition.description is not None
-            assert definition.id is not None
-            assert definition.name is not None
-            assert len(definition.permissions)
-            assert definition.role_name is not None
-            assert definition.role_type is not None
-            assert definition.type is not None
-
-    @ResourceGroupPreparer(random_name_enabled=True)
-    @KeyVaultPreparer()
-    @AccessControlClientPreparer()
-    def test_role_assignment(self, client):
+    def test_role_definitions(self):
+        client = KeyVaultAccessControlClient(self.managed_hsm["url"], self.credential)
+
+        # list initial role definitions
+        scope = KeyVaultRoleScope.global_value
+        original_definitions = [d for d in client.list_role_definitions(scope)]
+        assert len(original_definitions)
+
+        # create custom role definition
+        definition_name = self.get_replayable_uuid("definition-name")
+        permissions = [KeyVaultPermission(allowed_data_actions=[DataActionPermission.read_hsm_key])]
+        created_definition = client.set_role_definition(
+            role_scope=scope, role_definition_name=definition_name, permissions=permissions
+        )
+        assert "/" in created_definition.assignable_scopes
+        assert created_definition.name == definition_name
+        assert len(created_definition.permissions) == 1
+        assert created_definition.permissions[0].allowed_data_actions == [DataActionPermission.read_hsm_key]
+
+        # update custom role definition
+        permissions = [
+            KeyVaultPermission(
+                allowed_data_actions=[], denied_data_actions=[DataActionPermission.read_hsm_key]
+            )
+        ]
+        updated_definition = client.set_role_definition(
+            role_scope=scope, role_definition_name=definition_name, permissions=permissions
+        )
+        assert len(updated_definition.permissions) == 1
+        assert len(updated_definition.permissions[0].allowed_data_actions) == 0
+        assert updated_definition.permissions[0].denied_data_actions == [DataActionPermission.read_hsm_key]
+
+        # assert that the created role definition isn't duplicated
+        matching_definitions = [
+            d for d in client.list_role_definitions(scope)
+            if d.id == updated_definition.id
+        ]
+        assert len(matching_definitions) == 1
+
+        # get custom role definition
+        definition = client.get_role_definition(role_scope=scope, role_definition_name=definition_name)
+        assert_role_definitions_equal(definition, updated_definition)
+
+        # delete custom role definition
+        deleted_definition = client.delete_role_definition(scope, definition_name)
+        assert_role_definitions_equal(deleted_definition, definition)
+
+        assert not any(
+            d for d in client.list_role_definitions(scope) if d.id == deleted_definition.id
```

Comment:
This doesn't need to change but just FYI, you could simplify it a little by generating bools:
```suggestion
            d.id == deleted_definition.id for d in client.list_role_definitions(scope)
```